Spanner/src/Session/CacheSessionPool.php

```diff
@@ -502,6 +502,7 @@ public function warmup()
      * exceed the maximum number of sessions available per node, please be sure
      * to check the return value of this method to be certain all sessions have
      * been deleted.
+     * @return bool Returns true if all provided sessions are deleted, false otherwise.
```

Comment:
So changing the return type seems OK in this case as the return value isn't used anywhere in the code. It's used like a fire and forget.

But that might be a BC(even if it is null to bool).

With that said, let's add a unit test, that tests for both return types?

Reply:
Do we have any evidence if users use these by themselves? If so, then we probably can't change the return type like this 🤔 
Let's check with @bshaffer on the guidelines here.

Reply:
I'm surprised that the breaking change detector didn't flag this one, but generally speaking, it's probably okay since nobody would be relying on `null` returning from this function.

The bigger concern I have is that to me, this function returning `false` would indicate that the operation failed. But it's unclear if that's the case ("Returns true if all provided sessions are deleted, false otherwise.")

Reply:
A `false` indeed indicates that the `delete from server` operation failed and some sessions may still be tracked by the server.

The method description above suggests: `Clear the cache and attempt to delete all sessions in the pool` which should be sufficient, IMO. Any suggestions on documentation otherwise?

Reply:
I would just change it to "false if some delete operations failed" (or whatever it means for it to return false)

Is this something that needs to change to satisfy the feature, or for testability? Or are we changing it for better DX?

Reply:
Correct, its for DX. Exhausting all sessions leads to poorer performance also.